pandas/core/internals/blocks.py

```diff
@@ -789,8 +789,7 @@ def replace(self, to_replace, value, inplace=False, filter=None,
         # try to replace, if we raise an error, convert to ObjectBlock and
         # retry
         try:
-            values, _, to_replace, _ = self._try_coerce_args(self.values,
-                                                             to_replace)
+            values, to_replace = self._try_coerce_args(self.values, to_replace)
```

Comment:
It's hard to tell, but a lot of this looks like it might be unrelated to the np_percentile_compat function.  Is that the case?  If so, that's great (simplifying try_coerce_args etc would be a big win), but should be done separately.